
project build/
diff --git a/build/core/combo/HOST_linux-x86.mk b/build/core/combo/HOST_linux-x86.mk
index 445c12f..4bd8cd0 100644
--- a/build/core/combo/HOST_linux-x86.mk
+++ b/build/core/combo/HOST_linux-x86.mk
@@ -61,6 +61,6 @@ HOST_GLOBAL_CFLAGS += \
 	-include $(call select-android-config-h,linux-x86)
 
 # Disable new longjmp in glibc 2.11 and later. See bug 2967937.
-HOST_GLOBAL_CFLAGS += -D_FORTIFY_SOURCE=0
+HOST_GLOBAL_CFLAGS += -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
 
 HOST_NO_UNDEFINED_LDFLAGS := -Wl,--no-undefined
diff --git a/build/core/combo/TARGET_linux-x86.mk b/build/core/combo/TARGET_linux-x86.mk
index a683a0e..6da52b2 100644
--- a/build/core/combo/TARGET_linux-x86.mk
+++ b/build/core/combo/TARGET_linux-x86.mk
@@ -104,11 +104,7 @@ TARGET_GLOBAL_CFLAGS += -fstack-protector
 TARGET_GLOBAL_CPPFLAGS += \
 			-fno-use-cxa-atexit
 
-ifeq ($(TARGET_ARCH_VARIANT),x86-atom)
-    TARGET_GLOBAL_CFLAGS += -march=atom -mstackrealign -DUSE_SSSE3 -DUSE_SSE2 -mfpmath=sse
-else
-    TARGET_GLOBAL_CFLAGS += -march=i686
-endif
+TARGET_GLOBAL_CFLAGS += -march=i486
 
 TARGET_GLOBAL_CFLAGS += -mbionic
 TARGET_GLOBAL_CFLAGS += -D__ANDROID__
diff --git a/build/core/combo/arch/arm/armv4t.mk b/build/core/combo/arch/arm/armv4t.mk
index abc8fa2..aefb9b0 100644
--- a/build/core/combo/arch/arm/armv4t.mk
+++ b/build/core/combo/arch/arm/armv4t.mk
@@ -8,12 +8,6 @@
 # not supported.
 #
 $(warning ARMv4t support is currently a work in progress. It does not work right now!)
-ARCH_ARM_HAVE_THUMB_SUPPORT := false
-ARCH_ARM_HAVE_THUMB_INTERWORKING := false
-ARCH_ARM_HAVE_64BIT_DATA := false
-ARCH_ARM_HAVE_HALFWORD_MULTIPLY := false
-ARCH_ARM_HAVE_CLZ := false
-ARCH_ARM_HAVE_FFS := false
 
 DEFAULT_TARGET_CPU := arm920t
 
diff --git a/build/core/combo/arch/arm/armv5te.mk b/build/core/combo/arch/arm/armv5te.mk
index 29aada6..88e57b7 100644
--- a/build/core/combo/arch/arm/armv5te.mk
+++ b/build/core/combo/arch/arm/armv5te.mk
@@ -1,12 +1,6 @@
 # Configuration for Linux on ARM.
 # Generating binaries for the ARMv5TE architecture and higher
 #
-ARCH_ARM_HAVE_THUMB_SUPPORT     := true
-ARCH_ARM_HAVE_FAST_INTERWORKING := true
-ARCH_ARM_HAVE_64BIT_DATA        := true
-ARCH_ARM_HAVE_HALFWORD_MULTIPLY := true
-ARCH_ARM_HAVE_CLZ               := true
-ARCH_ARM_HAVE_FFS               := true
 
 # Note: Hard coding the 'tune' value here is probably not ideal,
 # and a better solution should be found in the future.
diff --git a/build/core/combo/arch/arm/armv7-a-neon.mk b/build/core/combo/arch/arm/armv7-a-neon.mk
index 7b5b8ed..746bca2 100644
--- a/build/core/combo/arch/arm/armv7-a-neon.mk
+++ b/build/core/combo/arch/arm/armv7-a-neon.mk
@@ -1,16 +1,6 @@
 # Configuration for Linux on ARM.
 # Generating binaries for the ARMv7-a architecture and higher with NEON
 #
-ARCH_ARM_HAVE_THUMB_SUPPORT     := true
-ARCH_ARM_HAVE_FAST_INTERWORKING := true
-ARCH_ARM_HAVE_64BIT_DATA        := true
-ARCH_ARM_HAVE_HALFWORD_MULTIPLY := true
-ARCH_ARM_HAVE_CLZ               := true
-ARCH_ARM_HAVE_FFS               := true
-ARCH_ARM_HAVE_ARMV7A            := true
-ARCH_ARM_HAVE_VFP               := true
-ARCH_ARM_HAVE_VFP_D32           := true
-ARCH_ARM_HAVE_NEON              := true
 
 # Note: Hard coding the 'tune' value here is probably not ideal,
 # and a better solution should be found in the future.
diff --git a/build/core/combo/arch/arm/armv7-a.mk b/build/core/combo/arch/arm/armv7-a.mk
index 0ca3ec6..f6b69f0 100644
--- a/build/core/combo/arch/arm/armv7-a.mk
+++ b/build/core/combo/arch/arm/armv7-a.mk
@@ -1,14 +1,6 @@
 # Configuration for Linux on ARM.
 # Generating binaries for the ARMv7-a architecture and higher
 #
-ARCH_ARM_HAVE_THUMB_SUPPORT     := true
-ARCH_ARM_HAVE_FAST_INTERWORKING := true
-ARCH_ARM_HAVE_64BIT_DATA        := true
-ARCH_ARM_HAVE_HALFWORD_MULTIPLY := true
-ARCH_ARM_HAVE_CLZ               := true
-ARCH_ARM_HAVE_FFS               := true
-ARCH_ARM_HAVE_ARMV7A            := true
-ARCH_ARM_HAVE_VFP               := true
 
 # Note: Hard coding the 'tune' value here is probably not ideal,
 # and a better solution should be found in the future.
diff --git a/build/target/board/emulator/BoardConfig.mk b/build/target/board/emulator/BoardConfig.mk
index 9ab607a..5a0b549 100644
--- a/build/target/board/emulator/BoardConfig.mk
+++ b/build/target/board/emulator/BoardConfig.mk
@@ -9,4 +9,4 @@ TARGET_NO_KERNEL := true
 HAVE_HTC_AUDIO_DRIVER := true
 
 # no hardware camera
-USE_CAMERA_STUB := true
+USE_CAMERA_STUB := false
diff --git a/build/target/board/generic/BoardConfig.mk b/build/target/board/generic/BoardConfig.mk
index 70f8585..65b3101 100644
--- a/build/target/board/generic/BoardConfig.mk
+++ b/build/target/board/generic/BoardConfig.mk
@@ -21,9 +21,14 @@ TARGET_NO_KERNEL := true
 TARGET_ARCH_VARIANT := armv7-a
 TARGET_CPU_ABI := armeabi-v7a
 TARGET_CPU_ABI2 := armeabi
+ARCH_ARM_HAVE_ARMV7A := true
+ARCH_ARM_HAVE_NEON := false
+ARCH_ARM_HAVE_THUMB_SUPPORT := true
+ARCH_ARM_HAVE_TLS_REGISTER := true
+ARCH_ARM_HAVE_VFP := true
+ARCH_ARM_HAVE_VFP_D32 := false
 
-HAVE_HTC_AUDIO_DRIVER := true
-BOARD_USES_GENERIC_AUDIO := true
+BOARD_USES_GENERIC_AUDIO := false
 
 # no hardware camera
 USE_CAMERA_STUB := true
@@ -35,4 +40,4 @@ WITH_DEXPREOPT := true
 endif
 
 # Build OpenGLES emulation guest and host libraries
-BUILD_EMULATOR_OPENGL := true
+BUILD_EMULATOR_OPENGL := false
diff --git a/build/target/board/generic_armv5/BoardConfig.mk b/build/target/board/generic_armv5/BoardConfig.mk
index d8fa2fe..21a6b2d 100644
--- a/build/target/board/generic_armv5/BoardConfig.mk
+++ b/build/target/board/generic_armv5/BoardConfig.mk
@@ -19,5 +19,11 @@ include build/target/board/generic/BoardConfig.mk
 TARGET_ARCH_VARIANT :=
 TARGET_CPU_ABI := armeabi
 TARGET_CPU_ABI2 :=
+ARCH_ARM_HAVE_ARMV7A := false
+ARCH_ARM_HAVE_NEON := false
+ARCH_ARM_HAVE_THUMB_SUPPORT := true
+ARCH_ARM_HAVE_TLS_REGISTER := false
+ARCH_ARM_HAVE_VFP := false
+ARCH_ARM_HAVE_VFP_D32 := false
 
 WITH_DEXPREOPT := false
diff --git a/build/target/board/generic_x86/BoardConfig.mk b/build/target/board/generic_x86/BoardConfig.mk
index 9826973..e7ea787 100644
--- a/build/target/board/generic_x86/BoardConfig.mk
+++ b/build/target/board/generic_x86/BoardConfig.mk
@@ -8,12 +8,10 @@ TARGET_NO_BOOTLOADER := true
 TARGET_NO_KERNEL := true
 TARGET_CPU_ABI := x86
 TARGET_ARCH := x86
-TARGET_ARCH_VARIANT := x86-atom
+TARGET_ARCH_VARIANT := x86
 TARGET_PRELINK_MODULE := false
 
-# The IA emulator (qemu) uses the Goldfish devices
-HAVE_HTC_AUDIO_DRIVER := true
-BOARD_USES_GENERIC_AUDIO := true
+BOARD_USES_GENERIC_AUDIO := false
 
 # no hardware camera
 USE_CAMERA_STUB := true
@@ -27,4 +25,4 @@ ifeq ($(HOST_OS),linux)
 WITH_DEXPREOPT := true
 endif
 # Build OpenGLES emulation host and guest libraries
-BUILD_EMULATOR_OPENGL := true
+BUILD_EMULATOR_OPENGL := false
diff --git a/build/target/product/full.mk b/build/target/product/full.mk
index c563bcc..be2ccf0 100644
--- a/build/target/product/full.mk
+++ b/build/target/product/full.mk
@@ -20,10 +20,10 @@
 # entirely appropriate to inherit from for on-device configurations.
 
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
-$(call inherit-product, $(SRC_TARGET_DIR)/board/generic/device.mk)
+$(call inherit-product, $(SRC_TARGET_DIR)/board/generic_armv5/device.mk)
 
 # Overrides
 PRODUCT_NAME := full
-PRODUCT_DEVICE := generic
+PRODUCT_DEVICE := generic_armv5
 PRODUCT_BRAND := Android
 PRODUCT_MODEL := Full Android on Emulator

project device/samsung/crespo/
diff --git a/device/samsung/crespo/libaudio/Android.mk b/device/samsung/crespo/libaudio/Android.mk
index 726859b..43f2ad8 100644
--- a/device/samsung/crespo/libaudio/Android.mk
+++ b/device/samsung/crespo/libaudio/Android.mk
@@ -1,7 +1,5 @@
 LOCAL_PATH:= $(call my-dir)
 
-ifneq ($(filter crespo crespo4g,$(TARGET_DEVICE)),)
-
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES:= aplay.c alsa_pcm.c alsa_mixer.c
 LOCAL_MODULE:= aplay
@@ -49,5 +47,3 @@ ifeq ($(BOARD_HAVE_BLUETOOTH),true)
   LOCAL_CFLAGS += -DWITH_A2DP
 endif
 include $(BUILD_SHARED_LIBRARY)
-
-endif

project frameworks/base/
diff --git a/frameworks/base/core/java/android/app/ActivityManager.java b/frameworks/base/core/java/android/app/ActivityManager.java
index fe1e7d7..58ddb18 100644
--- a/frameworks/base/core/java/android/app/ActivityManager.java
+++ b/frameworks/base/core/java/android/app/ActivityManager.java
@@ -64,7 +64,7 @@ public class ActivityManager {
     static public int staticGetMemoryClass() {
         // Really brain dead right now -- just take this from the configured
         // vm heap size, and assume it is in megabytes and thus ends with "m".
-        String vmHeapSize = SystemProperties.get("dalvik.vm.heapsize", "16m");
+        String vmHeapSize = SystemProperties.get("dalvik.vm.heapsize", "10m");
         return Integer.parseInt(vmHeapSize.substring(0, vmHeapSize.length()-1));
     }
     
diff --git a/frameworks/base/core/jni/AndroidRuntime.cpp b/frameworks/base/core/jni/AndroidRuntime.cpp
index d9759bf..ebe7592 100644
--- a/frameworks/base/core/jni/AndroidRuntime.cpp
+++ b/frameworks/base/core/jni/AndroidRuntime.cpp
@@ -650,7 +650,7 @@ int AndroidRuntime::startVm(JavaVM** pJavaVM, JNIEnv** pEnv)
     //options[curOpt++].optionString = "-verbose:class";
 
     strcpy(heapsizeOptsBuf, "-Xmx");
-    property_get("dalvik.vm.heapsize", heapsizeOptsBuf+4, "16m");
+    property_get("dalvik.vm.heapsize", heapsizeOptsBuf+4, "10m");
     //LOGI("Heap size: %s", heapsizeOptsBuf);
     opt.optionString = heapsizeOptsBuf;
     mOptions.add(opt);
diff --git a/frameworks/base/core/res/res/values/config.xml b/frameworks/base/core/res/res/values/config.xml
index 779028e..801a83e 100755
--- a/frameworks/base/core/res/res/values/config.xml
+++ b/frameworks/base/core/res/res/values/config.xml
@@ -57,7 +57,7 @@
     <!-- Flag indicating whether the surface flinger is inefficient
          at performing a blur.  Used by parts of the UI to turn off
          the blur effect where it isn't worth the performance hit. -->
-    <bool name="config_sf_slowBlur">false</bool>
+    <bool name="config_sf_slowBlur">true</bool>
     
     <!-- The duration (in milliseconds) of a short animation. -->
     <integer name="config_shortAnimTime">150</integer>
@@ -73,7 +73,7 @@
     <integer name="config_radioScanningTimeout">0</integer>
 
     <!-- A product with no SD card == not removable. -->
-    <bool name="config_externalStorageRemovable" product="nosdcard">false</bool>
+    <bool name="config_externalStorageRemovable" product="nosdcard">true</bool>
     <!-- Configures whether the primary external storage device is
          removable.  For example, if external storage is on an SD card,
          it is removable; if it is built in to the device, it is not removable.
@@ -149,13 +149,13 @@
     
     <!-- Flag indicating whether the we should enable the automatic brightness in Settings.
          Software implementation will be used if config_hardware_auto_brightness_available is not set -->
-    <bool name="config_automatic_brightness_available">false</bool>
+    <bool name="config_automatic_brightness_available">true</bool>
 
     <!-- Don't name config resources like this.  It should look like config_annoyDianne -->
     <bool name="config_annoy_dianne">true</bool>
 
     <!-- If this is true, the screen will come on when you unplug usb/power/whatever. -->
-    <bool name="config_unplugTurnsOnScreen">false</bool>
+    <bool name="config_unplugTurnsOnScreen">true</bool>
     
     <!-- If this is true, the screen will fade off. -->
     <bool name="config_animateScreenLights">true</bool>
@@ -188,7 +188,7 @@
     <!-- Control whether being in the desk dock should enable accelerometer
          based screen orientation.  Note this should probably default to true
          like car dock, but we haven't had a chance to test it. -->
-    <bool name="config_deskDockEnablesAccelerometer">false</bool>
+    <bool name="config_deskDockEnablesAccelerometer">true</bool>
 
     <!-- Control whether being in the car dock should enable accelerometer based
          screen orientation.  This defaults to true because putting a device in
@@ -208,7 +208,7 @@
     <integer name="config_lidNavigationAccessibility">0</integer>
 
     <!-- Indicate whether the SD card is accessible without removing the battery. -->
-    <bool name="config_batterySdCardAccessibility">false</bool>
+    <bool name="config_batterySdCardAccessibility">true</bool>
 
     <!-- Vibrator pattern for feedback about a long screen/key press -->
     <integer-array name="config_longPressVibePattern">
@@ -369,7 +369,7 @@
     <integer name="config_datause_notification_type">2</integer>
 
     <!-- Enables SIP on WIFI only -->
-    <bool name="config_sip_wifi_only">true</bool>
+    <bool name="config_sip_wifi_only">false</bool>
 
     <!-- Boolean indicating if restoring network selection should be skipped -->
     <!-- The restoring is handled by modem if it is true-->
@@ -384,7 +384,7 @@
 
     <!-- If this value is true, Sms encoded as octet is decoded by utf8 decoder.
          If false, decoded by Latin decoder. -->
-    <bool name="config_sms_utf8_support">false</bool>
+    <bool name="config_sms_utf8_support">true</bool>
 
     <!-- If this value is true, The mms content-disposition field is supported correctly.
          If false, Content-disposition fragments are ignored -->
@@ -460,12 +460,12 @@
 
     <!-- Set to true if the RSSI should always display CDMA signal strength
          even on EVDO -->
-    <bool name="config_alwaysUseCdmaRssi">false</bool>
+    <bool name="config_alwaysUseCdmaRssi">true</bool>
 
     <!-- If this value is true, duplicate Source/Destination port fields
          in WDP header of some carriers OMADM wap push are supported.
          ex: MSGTYPE-TotalSegments-CurrentSegment
              -SourcePortDestPort-SourcePortDestPort-OMADM PDU
          If false, not supported. -->
-    <bool name="config_duplicate_port_omadm_wappush">false</bool>
+    <bool name="config_duplicate_port_omadm_wappush">true</bool>
 </resources>
diff --git a/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardware.cpp b/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardware.cpp
index 818774d..f6baeef 100644
--- a/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardware.cpp
+++ b/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardware.cpp
@@ -96,6 +96,31 @@ PixelFormat DisplayHardware::getFormat() const  { return mFormat; }
 uint32_t DisplayHardware::getMaxTextureSize() const { return mMaxTextureSize; }
 uint32_t DisplayHardware::getMaxViewportDims() const { return mMaxViewportDims; }
 
+static status_t selectConfigForPixelFormat(
+        EGLDisplay dpy,
+        EGLint * attrs,
+        PixelFormat format,
+        EGLConfig* outConfig)
+{
+    EGLConfig config = NULL;
+    EGLint numConfigs = -1, n=0;
+    eglGetConfigs(dpy, NULL, 0, &numConfigs);
+    EGLConfig* configs = new EGLConfig[numConfigs];
+    eglChooseConfig(dpy, attrs, configs, numConfigs, &n);
+    for (int i=0 ; i<n ; i++) {
+        EGLint nativeVisualId = 0;
+        eglGetConfigAttrib(dpy, configs[i], EGL_NATIVE_VISUAL_ID, &nativeVisualId);
+        if (nativeVisualId>0 && format == nativeVisualId) {
+            *outConfig = configs[i];
+            delete [] configs;
+            return NO_ERROR;
+        }
+    }
+    delete [] configs;
+    return NAME_NOT_FOUND;
+}
+
+
 void DisplayHardware::init(uint32_t dpy)
 {
     mNativeWindow = new FramebufferNativeWindow();
@@ -105,6 +130,9 @@ void DisplayHardware::init(uint32_t dpy)
     mRefreshRate = fbDev->fps;
 
     mOverlayEngine = NULL;
+    int format;
+    ANativeWindow * window = mNativeWindow.get();
+    window->query(window, NATIVE_WINDOW_FORMAT, &format);
     hw_module_t const* module;
     if (hw_get_module(OVERLAY_HARDWARE_MODULE_ID, &module) == 0) {
         overlay_control_open(module, &mOverlayEngine);
@@ -114,11 +142,13 @@ void DisplayHardware::init(uint32_t dpy)
     EGLint numConfigs=0;
     EGLSurface surface;
     EGLContext context;
+    EGLBoolean result;
+    status_t err;
 
     // initialize EGL
     EGLint attribs[] = {
-            EGL_SURFACE_TYPE,   EGL_WINDOW_BIT,
-            EGL_NONE,           0,
+            EGL_SURFACE_TYPE,       EGL_WINDOW_BIT,
+            EGL_NONE,               0,
             EGL_NONE
     };
 
@@ -139,9 +169,8 @@ void DisplayHardware::init(uint32_t dpy)
     eglInitialize(display, NULL, NULL);
     eglGetConfigs(display, NULL, 0, &numConfigs);
 
-    EGLConfig config;
-    status_t err = EGLUtils::selectConfigForNativeWindow(
-            display, attribs, mNativeWindow.get(), &config);
+    EGLConfig config = NULL;
+    err = selectConfigForPixelFormat(display, attribs, format, &config);
     LOGE_IF(err, "couldn't find an EGLConfig matching the screen format");
     
     EGLint r,g,b,a;
@@ -222,7 +251,11 @@ void DisplayHardware::init(uint32_t dpy)
      * Gather OpenGL ES extensions
      */
 
-    eglMakeCurrent(display, surface, surface, context);
+    result = eglMakeCurrent(display, surface, surface, context);
+    if (!result) {
+        LOGE("Couldn't create a working GLES context. check logs. exiting...");
+        exit(0);
+    }
 
     GLExtensions& extensions(GLExtensions::getInstance());
     extensions.initWithGLStrings(
diff --git a/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardwareBase.cpp b/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardwareBase.cpp
index 90865da..f6aca7b 100644
--- a/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardwareBase.cpp
+++ b/frameworks/base/services/surfaceflinger/DisplayHardware/DisplayHardwareBase.cpp
@@ -89,7 +89,6 @@ bool DisplayHardwareBase::DisplayEventThread::threadLoop()
       err = read(fd, &buf, 1);
     } while (err < 0 && errno == EINTR);
     close(fd);
-    LOGW_IF(err<0, "ANDROID_WAIT_FOR_FB_SLEEP failed (%s)", strerror(errno));
     if (err >= 0) {
         sp<SurfaceFlinger> flinger = mFlinger.promote();
         LOGD("About to give-up screen, flinger = %p", flinger.get());
@@ -104,7 +103,6 @@ bool DisplayHardwareBase::DisplayEventThread::threadLoop()
       err = read(fd, &buf, 1);
     } while (err < 0 && errno == EINTR);
     close(fd);
-    LOGW_IF(err<0, "ANDROID_WAIT_FOR_FB_WAKE failed (%s)", strerror(errno));
     if (err >= 0) {
         sp<SurfaceFlinger> flinger = mFlinger.promote();
         LOGD("Screen about to return, flinger = %p", flinger.get());

project hardware/libhardware/
diff --git a/hardware/libhardware/modules/gralloc/framebuffer.cpp b/hardware/libhardware/modules/gralloc/framebuffer.cpp
index fe57a8a..4b82f01 100644
--- a/hardware/libhardware/modules/gralloc/framebuffer.cpp
+++ b/hardware/libhardware/modules/gralloc/framebuffer.cpp
@@ -355,11 +355,8 @@ int fb_device_open(hw_module_t const* module, const char* name,
         if (status >= 0) {
             int stride = m->finfo.line_length / (m->info.bits_per_pixel >> 3);
             int format = (m->info.bits_per_pixel == 32)
-                         ? HAL_PIXEL_FORMAT_RGBX_8888
+                         ? (m->info.red.offset ? HAL_PIXEL_FORMAT_BGRA_8888 : HAL_PIXEL_FORMAT_RGBX_8888)
                          : HAL_PIXEL_FORMAT_RGB_565;
-#ifdef NO_32BPP
-            format = HAL_PIXEL_FORMAT_RGB_565;
-#endif
             const_cast<uint32_t&>(dev->device.flags) = 0;
             const_cast<uint32_t&>(dev->device.width) = m->info.xres;
             const_cast<uint32_t&>(dev->device.height) = m->info.yres;

project system/core/
diff --git a/system/core/rootdir/init.rc b/system/core/rootdir/init.rc
index 8227273..86f54a1 100644
--- a/system/core/rootdir/init.rc
+++ b/system/core/rootdir/init.rc
@@ -5,7 +5,7 @@ on init
 
 sysclktz 0
 
-loglevel 3
+loglevel 0
 
 # setup the global environment
     export PATH /sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin
@@ -31,9 +31,7 @@ loglevel 3
     mkdir /mnt 0775 root system
     mkdir /mnt/sdcard 0000 system system
 
-# Create cgroup mount point for cpu accounting
     mkdir /acct
-    mount cgroup none /acct cpuacct
     mkdir /acct/uid
 
 # Backwards Compat - XXX: Going away in G*
@@ -62,7 +60,7 @@ loglevel 3
     mkdir /mnt/obb 0700 root system
     mount tmpfs tmpfs /mnt/obb mode=0755,gid=1000
 
-    write /proc/sys/kernel/panic_on_oops 1
+    write /proc/sys/kernel/panic_on_oops 0
     write /proc/sys/kernel/hung_task_timeout_secs 0
     write /proc/cpu/alignment 4
     write /proc/sys/kernel/sched_latency_ns 10000000
@@ -70,9 +68,7 @@ loglevel 3
     write /proc/sys/kernel/sched_compat_yield 1
     write /proc/sys/kernel/sched_child_runs_first 0
 
-# Create cgroup mount points for process groups
     mkdir /dev/cpuctl
-    mount cgroup none /dev/cpuctl cpu
     chown system system /dev/cpuctl
     chown system system /dev/cpuctl/tasks
     chmod 0777 /dev/cpuctl/tasks
@@ -89,17 +85,9 @@ loglevel 3
     # 5.0 %
     write /dev/cpuctl/bg_non_interactive/cpu.shares 52
 
-on fs
-# mount mtd partitions
-    # Mount /system rw first to give the filesystem a chance to save a checkpoint
-    mount yaffs2 mtd@system /system
-    mount yaffs2 mtd@system /system ro remount
-    mount yaffs2 mtd@userdata /data nosuid nodev
-    mount yaffs2 mtd@cache /cache nosuid nodev
-
 on post-fs
     # once everything is setup, no need to modify /
-    mount rootfs rootfs / ro remount
+    mount rootfs rootfs / rw remount
 
     # We chown/chmod /data again so because mount is run as root + defaults
     chown system system /data
@@ -156,6 +144,7 @@ on post-fs
     chmod 0660 /data/misc/wifi/wpa_supplicant.conf
     mkdir /data/local 0771 shell shell
     mkdir /data/local/tmp 0771 shell shell
+    mkdir /data/system 0771 system system
     mkdir /data/data 0771 system system
     mkdir /data/app-private 0771 system system
     mkdir /data/app 0771 system system
@@ -278,13 +267,10 @@ on boot
 ## Daemon processes to be run by init.
 ##
 service ueventd /sbin/ueventd
-    critical
 
 service console /system/bin/sh
     console
     disabled
-    user shell
-    group log
 
 on property:ro.secure=0
     start console
@@ -298,16 +284,14 @@ on property:ro.kernel.qemu=1
     start adbd
 
 on property:persist.service.adb.enable=1
+    write /sys/class/android_usb/android0/functions adb
+    write /sys/class/android_usb/android0/enable 1
     start adbd
 
 on property:persist.service.adb.enable=0
     stop adbd
 
 service servicemanager /system/bin/servicemanager
-    user system
-    critical
-    onrestart restart zygote
-    onrestart restart media
 
 service vold /system/bin/vold
     socket vold stream 0660 root mount
@@ -321,61 +305,40 @@ service debuggerd /system/bin/debuggerd
 service ril-daemon /system/bin/rild
     socket rild stream 660 root radio
     socket rild-debug stream 660 radio system
-    user root
-    group radio cache inet misc audio sdcard_rw
 
 service zygote /system/bin/app_process -Xzygote /system/bin --zygote --start-system-server
     socket zygote stream 666
-    onrestart write /sys/android_power/request_state wake
-    onrestart write /sys/power/state on
-    onrestart restart media
-    onrestart restart netd
 
 service media /system/bin/mediaserver
-    user media
-    group system audio camera graphics inet net_bt net_bt_admin net_raw
     ioprio rt 4
 
 service bootanim /system/bin/bootanimation
-    user graphics
-    group graphics
     disabled
     oneshot
 
 service dbus /system/bin/dbus-daemon --system --nofork
     socket dbus stream 660 bluetooth bluetooth
-    user bluetooth
-    group bluetooth net_bt_admin
 
 service bluetoothd /system/bin/bluetoothd -n
     socket bluetooth stream 660 bluetooth bluetooth
     socket dbus_bluetooth stream 660 bluetooth bluetooth
     # init.rc does not yet support applying capabilities, so run as root and
     # let bluetoothd drop uid to bluetooth with the right linux capabilities
-    group bluetooth net_bt_admin misc
     disabled
 
 service hfag /system/bin/sdptool add --channel=10 HFAG
-    user bluetooth
-    group bluetooth net_bt_admin
     disabled
     oneshot
 
 service hsag /system/bin/sdptool add --channel=11 HSAG
-    user bluetooth
-    group bluetooth net_bt_admin
     disabled
     oneshot
 
 service opush /system/bin/sdptool add --channel=12 OPUSH
-    user bluetooth
-    group bluetooth net_bt_admin
     disabled
     oneshot
 
 service pbap /system/bin/sdptool add --channel=19 PBAP
-    user bluetooth
-    group bluetooth net_bt_admin
     disabled
     oneshot
 
@@ -388,20 +351,15 @@ service flash_recovery /system/etc/install-recovery.sh
 service racoon /system/bin/racoon
     socket racoon stream 600 system system
     # racoon will setuid to vpn after getting necessary resources.
-    group net_admin
     disabled
     oneshot
 
 service mtpd /system/bin/mtpd
     socket mtpd stream 600 system system
-    user vpn
-    group vpn net_admin net_raw
     disabled
     oneshot
 
 service keystore /system/bin/keystore /data/misc/keystore
-    user keystore
-    group keystore
     socket keystore stream 666
 
 service dumpstate /system/bin/dumpstate -s

project system/extras/
diff --git a/system/extras/su/su.c b/system/extras/su/su.c
index b87cece..4db0a19 100644
--- a/system/extras/su/su.c
+++ b/system/extras/su/su.c
@@ -42,7 +42,7 @@
 int main(int argc, char **argv)
 {
     struct passwd *pw;
-    int uid, gid, myuid;
+    int uid, gid;
 
     if(argc < 2) {
         uid = gid = 0;
@@ -57,13 +57,6 @@ int main(int argc, char **argv)
         }
     }
 
-    /* Until we have something better, only root and the shell can use su. */
-    myuid = getuid();
-    if (myuid != AID_ROOT && myuid != AID_SHELL) {
-        fprintf(stderr,"su: uid %d not allowed to su\n", myuid);
-        return 1;
-    }
-    
     if(setgid(gid) || setuid(uid)) {
         fprintf(stderr,"su: permission denied\n");
         return 1;
